ui/src/app/components/new-user-satisfaction-survey-modal.spec.tsx

```diff
@@ -47,81 +57,100 @@ describe(NewUserSatisfactionSurveyModal.name, () => {
     setValidNewUserSatisfactionSurveyData();
   });
 
-  const createShallowWrapper = ({
+  const renderTestComponent = ({
     onCancel = () => {},
     onSubmitSuccess = () => {},
     createSurveyApiCall = () => Promise.resolve(),
   } = {}) => {
-    return shallow(
+    return render(
       <NewUserSatisfactionSurveyModal
-        onCancel={onCancel}
-        onSubmitSuccess={onSubmitSuccess}
-        createSurveyApiCall={createSurveyApiCall}
+        {...{ onCancel, onSubmitSuccess, createSurveyApiCall }}
       />
     );
   };
 
-  const findSubmitButton = (wrapper: ShallowWrapper) => {
-    return wrapper.find({ children: 'Submit' });
+  const findSubmitButton = (component: RenderResult) => {
+    return component.getByRole('button', { name: 'submit' });
   };
 
   it('should disable the submit button until a satisfaction value is selected', () => {
     setNewUserSatisfactionSurveyData('satisfaction', undefined);
-    expect(
-      findSubmitButton(createShallowWrapper()).prop('disabled')
-    ).toBeTruthy();
+
+    let component = renderTestComponent();
+    expectButtonElementDisabled(findSubmitButton(component));
+    component.unmount();
 
     setNewUserSatisfactionSurveyData(
       'satisfaction',
       NewUserSatisfactionSurveySatisfaction.VERY_SATISFIED
     );
-    expect(
-      findSubmitButton(createShallowWrapper()).prop('disabled')
-    ).toBeFalsy();
+    component = renderTestComponent();
+    expectButtonElementEnabled(findSubmitButton(component));
```

Comment:
Do you need to re-render here?

Could you surround your expectButtonElementEnabled in a waitfor instead?

```suggestion
    await waitFor(() =>
      expectButtonElementEnabled(findSubmitButton(component));
    );
```

Reply:
nice - this is much better